tidy3d/components/validators.py

```diff
@@ -173,11 +175,14 @@ def objects_in_sim_bounds(cls, val, values):
         sim_size = values.get("size")
         sim_box = Box(size=sim_size, center=sim_center)
 
+        # Do a strict check, unless simulation is 0D along a dimension
+        strict_ineq = [size != 0 and strict_inequality for size in sim_size]
+
         for position_index, geometric_object in enumerate(val):
-            if not sim_box.intersects(geometric_object.geometry):
+            if not sim_box.intersects(geometric_object.geometry, strict_inequality=strict_ineq):
                 message = (
-                    f"'simulation.{field_name}[{position_index}]'"
-                    "is completely outside of simulation domain."
+                    f"'simulation.{field_name}[{position_index}]' "
+                    "is outside of the simulation domain."
```

Comment:
maybe "is completely outside of the interior of the simulation domain" just so people don't get confused